server/rest/rest.go

```diff
@@ -155,14 +155,13 @@ func WithAdditionalGRPCOpts(opts []grpc.DialOption) ServerConfigOption {
 func WithEmbeddedOAuth2Server(keyPath string, keyPassword string, saveOnCreate bool, opts ...oauth2.AuthorizationServerOption) ServerConfigOption {
 	return func(c *restConfig, sm *runtime.ServeMux) {
 		var publicURL string
-
 		oauthHost := viper.GetString(config.EmbeddedOAuth2ServerPublicURLFlag)
 
 		// Check if an embedded OAuth2 server public URL is given, otherwise take localhost.
 		if oauthHost == "" {
 			publicURL = fmt.Sprintf("http://localhost:%d/v1/auth", httpPort)
 		} else {
-			publicURL = fmt.Sprintf("%s:%d/v1/auth", oauthHost, httpPort)
```

Comment:
FIx in Clouditor main. Change to 
`publicURL = fmt.Sprintf("%s/v1/auth", oauthHost)`.